pkg/cacheutil/memcached_client.go

```diff
@@ -573,18 +573,28 @@ func (c *memcachedClient) getMultiSingle(ctx context.Context, keys []string) (it
 func (c *memcachedClient) sortKeysByServer(keys []string) []string {
 	bucketed := make(map[string][]string)
 
-	for _, key := range keys {
-		addr, err := c.selector.PickServer(key)
-		// If we couldn't determine the correct server, return keys in existing order
-		if err != nil {
-			return keys
-		}
+	addrs := *(addrsPool.Get().(*[]net.Addr))
```

Comment:
Could this logic for manipulating the addrsPool be in the selector rather than the client?